astropy/coordinates/sky_coordinate.py

```diff
@@ -362,7 +363,41 @@ def _parse_inputs(self, args, kwargs):
 
         return valid_kwargs
 
-    def transform_to(self, frame):
+    def transforms(self, other, accepting_defaults=False):
```

Comment:
I am not wed to the name `transforms`, but there should be some way to transform an `other` object into `self`'s frame.

Part of the problem may be that I don't know the design rationale for having both FK5(ra,dec) (which is a `BaseCoordinateFrame`) and SkyCoord(ra,dec,frame='fk5'), with somewhat different functionality and some duplicated code.

I don't like the name `accepting_defaults=` either, but `take_attrs_from='source'` is also confusing, since in some circumstances you would still take attributes from the destination.